py/obj.h

```diff
@@ -721,7 +721,8 @@ bool mp_obj_is_true(mp_obj_t arg);
 bool mp_obj_is_callable(mp_obj_t o_in);
 bool mp_obj_equal(mp_obj_t o1, mp_obj_t o2);
 
-static inline bool mp_obj_is_integer(mp_const_obj_t o) { return mp_obj_is_int(o) || mp_obj_is_type(o, &mp_type_bool); } // returns true if o is bool, small int or long int
+static inline bool mp_obj_is_bool(mp_const_obj_t o) { return (o == MP_ROM_FALSE) || (o == MP_ROM_TRUE) || mp_obj_is_type(o, &mp_type_bool); } // returns true if o is bool
```

Comment:
This probably doesn't need the `mp_obj_is_type()` test at all, just the test against the 2 possible values.

But, for builds where `MICROPY_OBJ_IMMEDIATE_OBJS` is disabled, it might be simpler (smaller code) to just have the `mp_obj_is_type()` test (which will work fine).

And for object representation A this function could do just: `return (uintptr_t)o & 0xf == 0xe;`

Reply:
Is there a reason for choosing MP_ROM_XXX over mp_const_xxx ?

Reply:
not particularly no, I fairly naively picked them from the commit that broke my code (part of the openmv codebase that uses `mp_obj_is_integer()`). 
I didn't fully inspect the changes in the commit to properly understand what was going one, this was kind of a commit and run once I tested the end use functionality as working. I hadn't seen the other issue tracking this at this stage.